test/parallel/xyk-pallet.API.pool.weightedPools.test.ts

```diff
@@ -0,0 +1,194 @@
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(liqId.toNumber(), 20),
+    Assets.mintNative(testUser1)
+  );
+});
+
+test("Check that we can get the list of promoted pools with proofOfStake.promotedPoolRewards data storage", async () => {
+  const poolWeight = await getPoolWeight(liqId);
+
+  expect(poolWeight).bnGt(BN_ZERO);
+});
+
+test("Validate that weight can be modified by using updatePoolPromotion AND only sudo can update weights", async () => {
+  const api = getApi();
+
+  const poolWeightBefore = await getPoolWeight(liqId);
+
+  await signTx(
+    api,
+    api.tx.proofOfStake.updatePoolPromotion(
+      liqId,
+      poolWeightBefore.div(new BN(2))
+    ),
+    testUser1.keyRingPair
+  ).then((result) => {
+    const eventResponse = getEventResultFromMangataTx(result);
+    expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+    expect(eventResponse.data).toEqual("UnknownError");
+  });
+
+  await promotePool(
+    sudo.keyRingPair,
+    liqId,
+    poolWeightBefore.div(new BN(2)).toNumber()
+  );
+
+  const poolWeightAfter = await getPoolWeight(liqId);
+
+  expect(poolWeightAfter).bnEqual(poolWeightBefore.div(new BN(2)));
+});
+
+test("Testing that the sum of the weights can be greater than 100", async () => {
+  const [token2] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue, defaultCurrencyValue],
+    sudo
+  );
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(token2, testUser1, Assets.DEFAULT_AMOUNT),
+    Assets.mintNative(testUser1),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token2,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    )
+  );
+
+  const liqId2 = await getLiquidityAssetId(MGA_ASSET_ID, token2);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(liqId2.toNumber(), 100),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.activateLiquidity(liqId, Assets.DEFAULT_AMOUNT.divn(2))
+    ),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.activateLiquidity(liqId2, Assets.DEFAULT_AMOUNT.divn(2))
+    )
+  );
+
+  //await waitForRewards(testUser1, liqId2);
+
+  const poolWeightLiq1 = await getPoolWeight(liqId);
+
+  const poolWeightLiq2 = await getPoolWeight(liqId2);
+
+  const sumPoolsWeights = poolWeightLiq1.add(poolWeightLiq2);
+  const ratioPoolsWeights = poolWeightLiq2.div(poolWeightLiq1);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqId)),
+    Sudo.sudoAs(testUser1, Xyk.claimRewardsAll(liqId2))
+  );
+
+  await waitForRewards(testUser1, liqId2);
+
+  const rewardsLiqId1 = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqId
+  );
+
+  const rewardsLiqId2 = await getRewardsInfo(
+    testUser1.keyRingPair.address,
+    liqId2
+  );
+
+  expect(sumPoolsWeights).bnGt(BN_HUNDRED);
+  expect(rewardsLiqId1.rewardsAlreadyClaimed.mul(ratioPoolsWeights)).bnLte(
+    rewardsLiqId2.rewardsAlreadyClaimed
```

Comment:
do not fully understand this asset.

Reply:
Here I am checking that the weight of one pool is relative to another. Unfortunately, the value is not exactly 5 times larger, so I use `bnLte `instead of `bnEqual`